### DISCUSSION TITLE
Multiple q-steps in single v-for

### DISCUSSION THREAD

**Discussion by jterblanche** [quasarframework/quasar#12255] [General - Components / Directives / etc] [answered]
Any pointers/ideas on how to add multiple q-steps to a q-stepper in a single v-for loop using a template?

Example 1 works but with Example 2 the stepper does contain any steps when viewed in browser.

_Example 1 (works):_
```
<template>
  <q-stepper v-model="currentStep" header-nav>
    <q-step v-for="(product, idx) in order" :key="idx" :name="`product_${idx}_step_1`" :title="`Locate ${product.code}`">
      <div>Step 1: Locate {{ product.code }}</div>
    </q-step>
  </q-stepper>
</template>
```

_Example 2 (does not work):_
```
<template>
  <q-stepper v-model="currentStep" header-nav>
    <template v-for="(product, idx) in order" :key="idx">
      <q-step :name="`product_${idx}_step_1`" :title="`Locate ${product.code}`">
        <div>Step 1: Locate {{ product.code }}</div>
      </q-step>
      <q-step :name="`product_${idx}_step_2`" :title="`Confirm ${product.code} Specs`">
        <div>Step 2: Confirm Product Specifications for {{ product.code }}</div>
      </q-step>
  </q-stepper>
</template>
```

``order`` is simply an array of products, implementation omitted for the sake of brevity.

**Comment by MilosPaunovic** [comment]
[https://codepen.io/milospaunovic/pen/eYepjwg?editors=1011](https://codepen.io/milospaunovic/pen/eYepjwg?editors=1011)

**Comment by jterblanche** [comment]
Thanks for your reply. Not quite the answer to my question, as my question relates to adding **_multiple steps per order_** inside  ``<template></template>``.

But - your answer has led me on to discover something else. Example 2 works with the stepper in vertical mode, but not in horizontal mode.
Why is it then that Example 2 works with the stepper in vertical mode, but not in horizontal?

**Comment by MilosPaunovic** [comment]
Then I'm not really sure what is the problem that's troubling you, so it would be best if you create a Codepen reproduction of the issue.

**Comment by jterblanche** [comment]
[https://codepen.io/jterblanche/pen/ZEabgZx](https://codepen.io/jterblanche/pen/ZEabgZx)

To see the issue in action, simply comment out any one of the two steps in the "Broken" example. It doesn't matter which one of the two is commented out.

**Reply by MilosPaunovic** [reply]
When I do comment one step in `broken` section, tabs are still clickable. What should I pay attention to?

**Comment by jterblanche** [comment]
When no step in broken is commented out, the stepper collapses and is not
visible anymore.

**Comment by MilosPaunovic** [comment]
Ohhh... Now I understand. Okay, there seems to be the problem when `QStep`s aren't direct descendants of `QStepper`. I'm not sure if that's by design, so we can't wait for confirmation from the core team.

In the meantime, is there any particular reason why you need to use `template` tags?

**Comment by jterblanche** [comment]
I don't need to use `template` if there's another tag I can use.

I do need multiple steps in a v-for for each order though, and they need to
be in the correct sequence.

If I do a `q-step` `v-for` for each step of each order, the sequence would look
like this, which is incorrect.

Order 1 step 1
Order 2 step 1
Order 3 step 1
Order 1 step 2
Order 2 step 2
Order 3 step 2
...

The reason I tried wrapping the `q-step`(s) in `template` is because the
correct sequence I need is this:
Order 1 step 1
Order 1 step 2
Order 2 step 1
Order 2 step 2
Order 3 step 1
Order 3 step 3

**Comment by MilosPaunovic** [comment]
Could you create a computed property which will return orders sorted the way you want 'em and loop through `QStep`s using that?

**Comment by jterblanche** [comment]
I doubt it but I'll give it a try tonight.

**Comment by rstoenescu** [comment, marked as answer]
Seems like an under the hood change in Vue was introduced, partially breaking one of Quasar's algorithm.
A fix for it will land in Quasar v2.5.0

**Comment by jterblanche** [comment]
Excellent thank you!
